diff --git a/src/auth/UbuntuSSO.ts b/src/auth/UbuntuSSO.ts
index 4501b42..c1166fe 100644
--- a/src/auth/UbuntuSSO.ts
+++ b/src/auth/UbuntuSSO.ts
@@ -9,7 +9,8 @@ import Enquirer = require("enquirer");
 import Puppeteer from "puppeteer";
 import { Ora } from "ora";
 import { CookieJar } from "tough-cookie";
-import { existsSync, writeFileSync } from "fs";
+import { existsSync, writeFileSync, mkdirSync } from "fs";
+import { dirname } from "path";
 
 export default class UbuntuSSO extends Authentication {
     private httpsAgent;
@@ -66,6 +67,10 @@ export default class UbuntuSSO extends Authentication {
     }
 
     private saveUserSettings() {
+        const confDir = dirname(this.config.userSettingsPath);
+        if (!existsSync(confDir)) {
+            mkdirSync(confDir, { recursive: true });
+        }
         writeFileSync(this.config.userSettingsPath, JSON.stringify(this.jar));
     }
 
diff --git a/src/config/Config.ts b/src/config/Config.ts
index 86971b9..c421390 100644
--- a/src/config/Config.ts
+++ b/src/config/Config.ts
@@ -86,8 +86,8 @@ class Config {
         return join(
             homedir(),
             this.isCanonical()
-                ? ".canonical-greenhouse.json"
-                : ".ght-greenhouse.json"
+                ? ".config/ght/.canonical-greenhouse.json"
+                : ".config/ght/.ght-greenhouse.json"
         );
     }
 
