diff --git a/craft_providers/bases/buildd.py b/craft_providers/bases/buildd.py
index f959629..14f7af9 100644
--- a/craft_providers/bases/buildd.py
+++ b/craft_providers/bases/buildd.py
@@ -579,15 +579,16 @@ class BuilddBase(Base):
                     ) from error
             else:
                 try:
-                    snap_installer.inject_from_host(
+                    snap_installer.install_from_store(
                         executor=executor,
                         snap_name=snap.name,
                         classic=snap.classic,
+                        channel="2.2/stable"
                     )
                 except snap_installer.SnapInstallationError as error:
                     raise BaseConfigurationError(
                         brief=(
-                            f"failed to inject host's snap {snap.name!r} "
+                            f"failed to install charmcraft snap {snap.name!r} "
                             "into target environment."
                         )
                     ) from error
